taskcluster/mozillavpn_taskgraph/actions/release_promotion.py

```diff
@@ -0,0 +1,147 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from taskgraph.actions.registry import register_callback_action
+from taskgraph.decision import taskgraph_decision
+from taskgraph.parameters import Parameters
+from taskgraph.taskgraph import TaskGraph
+from taskgraph.util.taskcluster import get_artifact
+from taskgraph.util.taskgraph import (
+    find_decision_task,
+    find_existing_tasks_from_previous_kinds,
+)
+
+RELEASE_PROMOTION_PROJECTS = (
+    "https://github.com/mozilla-mobile/mozilla-vpn-client",
+    "https://github.com/mozilla-releng/staging-mozilla-vpn-client",
+)
+
+
+def is_release_promotion_available(parameters):
+    return parameters["head_repository"] in RELEASE_PROMOTION_PROJECTS
+
+
+@register_callback_action(
+    name="release-promotion",
+    title="Release Promotion",
+    symbol="${input.release_promotion_flavor}",
+    description="Promote or ship the Mozilla VPN client or its addons.",
+    generic=False,
+    order=500,
+    context=[],
+    available=is_release_promotion_available,
+    schema=lambda graph_config: {
+        "type": "object",
+        "properties": {
+            "build_number": {
+                "type": "integer",
+                "default": 1,
+                "minimum": 1,
+                "title": "The release build number",
+                "description": (
+                    "The release build number. Starts at 1 per "
+                    "release version, and increments on rebuild."
+                ),
+            },
+            "do_not_optimize": {
+                "type": "array",
+                "description": (
+                    "Optional: a list of labels to avoid optimizing out "
+                    "of the graph (to force a rerun)."
+                ),
+                "items": {
+                    "type": "string",
+                },
+            },
+            "revision": {
+                "type": "string",
+                "title": "Optional: revision to ship",
+                "description": ("Optional: the revision to ship."),
+            },
```

Comment:
Please remove this as it looks like it isn't used. And I agree that we shouldn't use it, being able to ship off a revision other than the current one feels like a bad feature to have.

Reply:
Done :)